Electronique/main.py

```diff
@@ -0,0 +1,60 @@
+from gpiozero import MotionSensor
+from picamera2 import Picamera2
+import time
+import cv2
+import os
+
+# Configuration de la caméra
+camera = Picamera2()
+
+# Configuration des capteurs de mouvement
+pir_out = MotionSensor(4)  # Capteur OUT
+pir_in = MotionSensor(27)  # Capteur IN
+
+try:
+    print("Attente du mouvement. Appuyez sur Ctrl+C pour quitter.")
+    time.sleep(2)  # Attend que le capteur s'initialise
+
+    while True:
+        if pir_out.motion_detected or pir_in.motion_detected:
+            print("Mouvement détecté! Enregistrement de la vidéo.")
+
+            # Détermine quel capteur a détecté le mouvement en premier
+            sensor_trigger = "in" if pir_in.motion_detected else "out"
+
+            # Récupère la date et l'heure pour le nom de fichier
+            timestamp = time.strftime("%d-%m-%y-%H-%M-%S", time.localtime())
+            filename = f"{sensor_trigger}-{timestamp}.mp4"
```

Comment:
Remarque 3 : Je comprends mieux comment vous faisiez pour pouvoir savoir si c'était des captures d'entrées ou de sorties du chat (supposé) détecté par la caméra, vous utilisez la variable sensor_trigger qui sera soit "in" si le capteur d'entrée détecte quelque chose sinon c'est "out" pour tout autre détection que celle du capteur "in" donc le capteur "out" d'office. Puis vous repreniez cette variable pour la mettre dans le nom de fichier mp4. Astucieux !

Reply:
Tout à fait!